src/classes/ADDR_Contact_TEST.cls

```diff
@@ -0,0 +1,1112 @@
+            addrs[i].Parent_Contact__c = contactsAddrs.contacts[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'override' + i;
+            addrs[i].MailingCity__c = 'override' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        
+        // set the Contacts' address overrides
+        for (Integer i = 0; i < contactsAddrs.contacts.size(); i++) {
+            Contact contact = contactsAddrs.contacts[i];
+            contact.Current_Address__c = addrs[i].Id;
+            contact.is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update contactsAddrs.contacts;        
+        
+        // now let's delete the override Addresses
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        delete addrs;
+        Test.stopTest();
+    
+        // verify that the Contact address fields went back to the default
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
+        MailingPostalCode, MailingCountry, MailingLatitude, MailingLongitude, is_Address_Override__c, Current_Address__c from Contact]);
+        
+        for (Contact contact : mapAccIdAcc.values()) {
+            system.assert(contact.MailingStreet.contains('Street'));
+            system.assert(contact.MailingCity.contains('City')); 
+            system.assertEquals(false, contact.is_Address_Override__c);
+            system.assertNotEquals(null, contact.Current_Address__c);                
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Making an Address non-default. That is, the address is no longer the default one for the Contact.  
+    verify: no change to Contact address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateNonDefaultAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Contact
+        UTIL_UnitTestData_TEST.ContactsWithAddrs contactsAddrs = UTIL_UnitTestData_TEST.createTestContactsAddrs(2);
+
+        // now let's update the Addresses
+        for (Integer i = 0; i < contactsAddrs.addrs.size(); i++) {
+            Address__c addr = contactsAddrs.addrs[i];
+            addr.Default_Address__c = false;
+        }
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        update contactsAddrs.addrs;
+        Test.stopTest();
+    
+        // verify that the Contact and Contacts don't share the same address and it's new for the Contact!
+        map<Id, Contact> mapAccIdAcc = new map<Id, Contact>([select Id, Name, MailingStreet, MailingCity, MailingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.